inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -24,8 +25,12 @@ def __init__(self, db_file=UDM_DB_FILE) -> None:
         """
         self._db_file = db_file
         if self._db_file != ":memory:" and not os.path.exists(self._db_file):
-            raise DispatcherException(f"Error: Database file {self._db_file} does not exist.")
-
+            # Create database file if not exist
+            logger.info(f"Database file doesn't exist. Creating the file.")
+            file_descriptor = os.open(self._db_file, os.O_CREAT)
+            # Set permission of the file (rw for owner and group)
+            os.chmod(self._db_file, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IWGRP)
+            os.close(file_descriptor)
         try:
             with sqlite3.connect(self._db_file) as conn:
```

Comment:
<h3>$\color{#FFA500}Improvements: Performance  Optimization$</h3>

The database connection is attempted to be established in the `__init__` method but is not used until later. Consider lazy initialization of the database connection to improve efficiency and resource usage, especially if the database might not be needed immediately upon object instantiation.